odoo/addons/base/models/res_lang.py

```diff
@@ -209,77 +240,64 @@ def install_lang(self):
                 partner.write({'lang': lang_code})
         return True
 
-    @tools.ormcache('code')
-    def _lang_get_id(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).id
-
-    @tools.ormcache('code')
-    def _lang_get_direction(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).direction
-
-    @tools.ormcache('url_code')
-    def _lang_get_code(self, url_code):
-        return self.with_context(active_test=True).search([('url_code', '=', url_code)]).code or url_code
+    # ------------------------------------------------------------
+    # cached methods for **active** languages
+    # ------------------------------------------------------------
+    @property
+    def CACHED_FIELDS(self) -> list[str]:
+        """ Return the fields to cache for the active languages
+        Please make sure all these fields don't depend on other models or context and are not translated
```

Comment:
Nitpicking but for the docstrings (in all the PR), I'd try to line return at max 80 char?